### DISCUSSION TITLE
[IDEA] Extract form schema, table columns to a seperate classes

### DISCUSSION THREAD

**Discussion by 4m1n0s** [filamentphp/filament#6366] [Ideas]
Hey,

Issue:
sometimes when you deal with large forms, or table columns this will make the component so large and big

the idea is to extract form schema, table columns to a seperate class, and make $form->schema() accept Arrayable and call toArray or Schema interface and the seperated class will be responsible only for schema, columns

something like that:

<img width="827" alt="image" src="https://user-images.githubusercontent.com/2763884/235387490-0a418459-1884-402f-a116-a9b0f3330b90.png">

**Comment by danharrin** [comment]
You can do this however you want in your own project. Nothing needs to be changed in the Filament core.

You could create a class with a static method that returns the form schema array. Then pass that directly to `schema()`.

Or you could create a class with a static method that accepts `$form` and applies the changes, then you just call `ClassName::form($form)`
